From 2ab84ef10562be99e01e2407daaef8bdaf2a89e8 Mon Sep 17 00:00:00 2001
From: jiangxin <jiangxin@hygon.cn>
Date: Tue, 24 Aug 2021 17:31:28 +0800
Subject: [PATCH 6/9] anolis: cpu/i386: populate CPUID 0x8000_001F when CSV is
 active

On Hygon platform, bit 30 of EAX indicates whether
this feature is supported in hardware.

When CSV is active, CPUID 0x8000_001F provides
information for it.

Signed-off-by: Xin Jiang <jiangxin@hygon.cn>
Change-Id: I19322a73d0df091c0bbb4ca2a3cc5f4225cc2c9f
---
 target/i386/cpu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/target/i386/cpu.c b/target/i386/cpu.c
index 0f71ff9fea..f8e7498971 100644
--- a/target/i386/cpu.c
+++ b/target/i386/cpu.c
@@ -28,6 +28,7 @@
 #include "sysemu/hvf.h"
 #include "kvm/kvm_i386.h"
 #include "sev.h"
+#include "csv.h"
 #include "qapi/error.h"
 #include "qapi/qapi-visit-machine.h"
 #include "qapi/qmp/qerror.h"
@@ -5853,6 +5854,7 @@ void cpu_x86_cpuid(CPUX86State *env, uint32_t index, uint32_t count,
         if (sev_enabled()) {
             *eax = 0x2;
             *eax |= sev_es_enabled() ? 0x8 : 0;
+            *eax |= csv_enabled() ? 0x40000000 : 0; /* bit 30 for CSV */
             *ebx = sev_get_cbit_position();
             *ebx |= sev_get_reduced_phys_bits() << 6;
         }
-- 
2.17.1

